### PR TITLE
Tyrone/sign in/test

### PR DESCRIPTION
**Pull request by Ty-Won**
Added the tests to accompany the login endpoint.

### CODE REVIEW

File: backend/test/users_test.js [hunk 1]
```diff
@@ -734,3 +737,101 @@ describe("Test get student completed courses", () => {
...
+    await users.insertStudentUser(
+      "team",
+      "primus1234",
+      "testing@gmail.com",
+      250502459
+    );
+  });
+
+  it(`Empty username responds with ${expectedUsernameNotEmpty}`, function(done) {
+    users.login(null, "primus1234").catch(error => {
```
Comment (zerfuth) [inline backend/test/users_test.js]: Is null username the same as empty username? Like if you put in "" versus null, would it behave the same?

File: backend/test/users_test.js [hunk 2]
```diff
@@ -734,3 +737,101 @@ describe("Test get student completed courses", () => {
...
+  it(`Empty password responds with ${expectedPasswordNotEmpty}`, function(done) {
+    users.login("team", null).catch(error => {
+      return new Promise(function(resolve) {
+        assert.equal(error.message, expectedPasswordNotEmpty);
+        resolve();
+      }).then(done);
+    });
+  });
+
+  it(`Empty fields responds with ${expectedPasswordNotEmpty}`, function(done) {
```
Comment (zerfuth) [inline backend/test/users_test.js]: Did you mean responds with ${expectedUsernameNotEmpty} ?

File: backend/test/users_test.js [hunk 3]
```diff
@@ -734,3 +737,101 @@ describe("Test get student completed courses", () => {
   });
 });
 
+describe("Testing Login", () => {
+  before(async () => {
+    await users.insertStudentUser(
+      "team",
+      "primus1234",
+      "testing@gmail.com",
+      250502459
```
Comment (si-x-li) [inline backend/test/users_test.js]: Consider making these entries into variables so that we only have to modify at 1 location

File: backend/test/users_test.js [hunk 4]
```diff
@@ -734,3 +737,101 @@ describe("Test get student completed courses", () => {
...
+    users.login(null, null).catch(error => {
+      return new Promise(function(resolve) {
+        assert.equal(error.message, expectedUsernameNotEmpty);
+        resolve();
+      }).then(done);
+    });
+  });
+
+  it(`Non existing Username responds with ${expectedNonExistentUser}`, function(done) {
+    users.login("teamm", "primus1234").catch(error => {
```
Comment (si-x-li) [inline backend/test/users_test.js]: Same as above use variables instead

File: backend/test/users_test.js [hunk 5]
```diff
@@ -734,3 +737,101 @@ describe("Test get student completed courses", () => {
...
+      return new Promise(function(resolve) {
+        assert.equal(error.message, expectedNonExistentUser);
+        resolve();
+      }).then(done);
+    });
+  });
+
+  it(`Incorrect Password responds with ${expectedIncorrectPassword}`, function(done) {
+    users.login("team", "primus123456789").catch(error => {
```
Comment (si-x-li) [inline backend/test/users_test.js]: Same here

File: backend/test/users_test.js (hunk 1 above)
Comment (si-x-li) [reply backend/test/users_test.js]: I think it wouldn't behave the same since only null are caught in the validation

Comment (Ty-Won) [reply backend/test/users_test.js]: Good catch, "" != null, the condition checks for 
userInfo == undefined || userInfo.length == 0
so it will catch it. I will add in a test for empty instead of null

File: backend/test/users_test.js (hunk 2 above)
Comment (Ty-Won) [reply backend/test/users_test.js]: thanks!

File: backend/test/users_test.js [hunk 6]
```diff
@@ -734,3 +737,101 @@ describe("Test get student completed courses", () => {
...
+    users.login("team!$", "primus1234").catch(error => {
+      return new Promise(function(resolve) {
+        assert.equal(error.message, expectedInvalidUsername);
+        resolve();
+      }).then(done);
+    });
+  });
+
+  it(`Long username greater than 64 length responds with ${expectedUsernameLessThan64}`, function(done) {
+    users.login("teammmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmm", "primus").catch(error => {
```
Comment (KevenLYF) [inline backend/test/users_test.js]: lol
